app/src/main/jni/algorithm_module.hpp

```diff
@@ -45,6 +45,9 @@ class AlgorithmModule {
     virtual std::string status() const { return ""; }
     virtual int trackingStatus() { return -1; };
 
+    virtual void writeInfoFile(const std::string &mode, const std::string &device) {};
+    virtual void writeParamsFile(float focalLength) {};
```

Comment:
In any case, there should be no need for this generic algorithm module API to know anything about other custom parameters, especially if operating in recording-only mode (there should be no algo parameters then, just device model/parameter info)